Modules/Filtering/LabelMap/include/itkStatisticsLabelMapFilter.hxx

```diff
@@ -73,8 +73,10 @@ StatisticsLabelMapFilter<TImage, TFeatureImage>::ThreadedProcessLabelObject(Labe
   typename HistogramType::MeasurementVectorType featureImageMax(1);
 
 
-  if (NumericTraits<typename Self::FeatureImagePixelType>::IsInteger &&
-      m_NumberOfBins == 1 << (8 * sizeof(typename Self::FeatureImagePixelType)))
+  constexpr size_t short bitsShift =
+    std::min(8 * sizeof(typename Self::FeatureImagePixelType), 8 * sizeof(unsigned int) - 1);
+  if (NumericTraits<typename Self::FeatureImagePixelType>::IsInteger && sizeof(FeatureImagePixelType) <= 2 &&
+      m_NumberOfBins == 1 << bitsShift)
```

Comment:
I assume that this pull request is needed because there is a system where `sizeof(unsigned int)` is only 2.  In such a case and only in such a case, this pull request uses a bitsShift of 15 (rather than 16 as previously) in the case that `sizeof(FeatureImagePixelType)` is also 2.  Unfortunately, the goal is to have 2<sup>16</sup> buckets in such a case, centered on the integer values.  Instead we will have only 2<sup>15</sup> buckets.  I guess the padding / centering is still appropriate....

However, I think the best solution would be to use 2<sup>16</sup> buckets in such a case.  Can that be accomplished on every system by using `unsigned long` in key places?

Reply:
> I assume that this pull request is needed because there is a system where `sizeof(unsigned int)` is only 2.
No. It occurred with when the pixel type was 64-bits, and the size of m_NumberOfBins was only 32-bits.

Reply:
Correct me if I am wrong, but the problem is that in the old code
```c++
    return NumericTraits<FeatureImagePixelType>::IsInteger && sizeof(FeatureImagePixelType) <= 2
             ? 1 << (8 * sizeof(FeatureImagePixelType))
             : 128;
```
the `1 << (8 * sizeof(FeatureImagePixelType))` was causing a compiler warning even when code was choosing the `128` branch.

Reply:
Correct.